<a href="https://cognitiveclass.ai"><img src = "https://ibm.box.com/shared/static/9gegpsmnsoo25ikkbl4qzlvlyjbgxs5x.png" width = 400> </a>

<h1 align=center><font size = 5>Capstone project: Battle of the neighborhoods</font></h1>


## Recommendation of a new Government Medical college Hospital, Apartment for doctors, Hostel for Students in Kasaragod district


## Table of Contents

<div class="alert alert-block alert-info" style="margin-top: 20px">

<font size = 3>

1. <a href="#item1">Introduction:Business Problem</a>

2. <a href="#item2">Data Resources</a>

3. <a href="#item3">Methodology</a>

4. <a href="#item4">Download and Explore Dataset</a>

5. <a href="#item5">Explore Venues in Kasaragod</a>

6. <a href="#item6"> Analyze Each Locations in Kasaragod District</a>

7. <a href="#item7"> Cluster Locations in Kasaragod District</a>

8. <a href="#item8"> Examine Clusters</a>

9. <a href="#item9"> Results</a>






</font>
</div>

<a id='item1'></a>

## 1. Introduction:Business Problem

  Kasaragod district is one of the 14 districts in the southern Indian state of Kerala. Kasaragod became part of Kannur district following the reorganisation of states due to predominance of Malayalam speakers. Kasaragod was declared as a district on 24 May 1984. At the same time, Kanyakumari district was separated from Kerala. Read more about Kasargod district here:-https://en.wikipedia.org/wiki/Kasaragod_district.

  The endosulfan tragedy which occurred in Kasaragod district is now well known. It is considered by many experts in the field of pesticide toxicity as one of the worst pesticide disasters to happen to a region and its people. Still, the victims are living in Kasaragod district. You can read from here https://www.cambridge.org/core/journals/international-annals-of-criminology/article/environmental-crime-and-victimization-a-green-criminological-analysis-of-the-endosulfan-disaster-kasargod-kerala/1E6ECAF4A29CA2D9538133DB86DDCC1A

  During Monsoon season diseases like Chikungunya, Dengue, Rat fever, Viral fever are very common among People. They depend mainly to Government District Medical Hospital, Chemmattamvayal, Kasaragod. But it lacks advanced medical facilities and shortage of efficient doctors. Diagnosing centres present in the different hospitals of the district lack to diagnose Virus-like H1N1, Nipah, Zika Virus etc. We used to send the sample to the Manipal Institute of Virology. Due to a large number of samples from different parts. Put our sample in a queue. Around 4-5 days requires to diagnose these type of Viruses. Above middle-class family members depends on Private Hospitals. The situation of private hospitals is also similar. If the condition of the patient is severe, they prescribe to any Private Hospital in Mangalore.

  We lack advanced medical facility in our district. A Government medical college will provide better treatment for Endosulfan victims as well as people suffering serious health issues. The medical college will transform the image of our district.
Very fewer hospitals are available in our district. We depend on Private hospitals in Mangaluru City. So, Here, I will find the most suitable place for a Government medical college. I have to find a suitable place in Kasaragod district in such a way that Medical students and Doctors in the newly established Hospital should access basic necessities.

<a id='item2'></a>

## 2. Data Resources

I used the following site for extracting the Pincode of different locations of Kasaragod district. 

https://data.gov.in/resources/all-india-pincode-directory-contact-details-along-latitude-and-longitude. 

It is given that Latitude and longitude information was available. You can verify that it is blank here. I use the geocoder Library and Google map API to extract the Latitude and Longitude information. After that, I saved the file in .csv format. I uploaded the file in my GitHub Repository. I am providing the link of my repository below.

https://github.com/maneeshnarayanan1993/Battle-of-Neighbourhood/blob/master/Kasaragod_Lat_Long.csv.

I will use the FourSquare API to explore different Location in Kasaragod. The Foursquare explore function will be used to get the most popular venues in each location.

<a id='item3'></a>

## 3. Methodology

The postal codes of different Location in Kasaragod will be collected https://data.gov.in website. Data will be processed using Microsoft Office Excel. After that, the geographical coordinates, latitude, and longitude will be retrieved by using the Google API. The Location with name, postal code, geographical coordinates will be regrouped and cleaned by Pandas library. Then FourSquare API will be called to retrieve all the venues in different locations. Finally, the machine learning algorithm K-means clustering would be applied to form the clusters of different venues categories for all the Location.

<a id='item4'></a>

## 4. Download and Explore Dataset

### Loading Geospatial data of Kasaragod District, Kerala, India

In [4]:

import types
import pandas as pd
from botocore.client import Config
import ibm_boto3

def __iter__(self): return 0

# @hidden_cell
# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.
# You might want to remove those credentials before you share the notebook.
client_875c8f21c0ab48679dce67bb76a49cc9 = ibm_boto3.client(service_name='s3',
    ibm_api_key_id='iqI-YWn1zcDve-qXaLQGgYWleInhkSnecnfyRLCRfQhV',
    ibm_auth_endpoint="https://iam.eu-gb.bluemix.net/oidc/token",
    config=Config(signature_version='oauth'),
    endpoint_url='https://s3.eu-geo.objectstorage.service.networklayer.com')

body = client_875c8f21c0ab48679dce67bb76a49cc9.get_object(Bucket='battleneighbourhood-donotdelete-pr-ssbhzuigxpfx8r',Key='Kasaragod_Lat_Long.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )
df_data_0=pd.read_csv(body)
# If you are reading an Excel file into a pandas DataFrame, replace `read_csv` by `read_excel` in the next statement.df_data_0 = pd.read_csv(body)
df_data_0


Location  Pincode   State  District   Latitude  Longitude
0          Achanthuruthi   671351  Kerala  Kasargod  12.229580  75.123890
1              Achikanam   671531  Kerala  Kasargod  12.332950  75.133900
2             Adkathbail   671121  Kerala  Kasargod  12.514748  74.977047
3                   Adur   671543  Kerala  Kasargod  12.557546  75.180804
4                 Ajanur   671531  Kerala  Kasargod  12.329081  75.074803
5               Alampady   671123  Kerala  Kasargod  12.534245  75.040736
6                 Ammeri   671322  Kerala  Kasargod  12.714228  74.999097
7           Anandashrama   671531  Kerala  Kasargod  12.339283  75.111810
8               Anekallu   671323  Kerala  Kasargod  12.752339  74.993968
9           Angadimogaru   671321  Kerala  Kasargod  12.635864  75.014521
10               Arikady   671321  Kerala  Kasargod  12.611301  74.947181
11             Ariyapady   671551  Kerala  Kasargod  12.508002  74.988211
12            Attenganam   671531  Kerala  Kasargod  12.389619  75.196852
13                Badaje   671323  Kerala  Kasargod  12.731822  74.906262
14                 Badur   671321  Kerala  Kasargod  12.658037  75.031909
15                 Balal   671533  Kerala  Kasargod  12.392013  75.265406
16            Balanthode   671532  Kerala  Kasargod  12.459245  75.320811
17             Balemoole   671552  Kerala  Kasargod  12.668430  75.144999
18                 Balla   671531  Kerala  Kasargod  12.320475  75.106612
19              Bandadka   671541  Kerala  Kasargod  12.501006  75.268323
20      Bangramanjeshwar   671323  Kerala  Kasargod  12.709134  74.900416
21                  Bare   671319  Kerala  Kasargod  12.447905  75.042131
22                 Bayar   671348  Kerala  Kasargod  12.691564  75.031909
23               Bedadka   671541  Kerala  Kasargod  12.456411  75.169131
24              Bedradka   671124  Kerala  Kasargod  12.450110  75.150031
25                 Bekal   671318  Kerala  Kasargod  12.420353  75.023146
26                 Bekur   671322  Kerala  Kasargod  12.672056  74.935491
27                  Bela   671321  Kerala  Kasargod  12.569345  75.043592
28               Belluru   671543  Kerala  Kasargod  12.586296  75.170590
29              Bengalam   671314  Kerala  Kasargod  12.293883  75.153036
..                   ...      ...     ...       ...        ...        ...
198            Thayanoor   671531  Kerala  Kasargod  12.351741  75.192476
199              Thekkil   671541  Kerala  Kasargod  12.488126  75.050645
200         Thekkilferry   671545  Kerala  Kasargod  12.492447  75.053083
201             Thuruthi   671351  Kerala  Kasargod  12.218447  75.123890
202  Timiri - chervathur   671313  Kerala  Kasargod  12.226383  75.177296
203            Trikarpur   671310  Kerala  Kasargod  12.141781  75.172049
204  Trikarpur Kadapuram   671310  Kerala  Kasargod  12.141781  75.172049
205      Trikarpur South   671311  Kerala  Kasargod  12.141781  75.172049
206               Udinur   671349  Kerala  Kasargod  12.160777  75.163294
207                 Udma   671319  Kerala  Kasargod  12.436750  75.017303
208         Udumbanthala   671350  Kerala  Kasargod  12.107254  75.176426
209             Udyavara   671323  Kerala  Kasargod  13.311580  74.732703
210            Ukkinadka   671552  Kerala  Kasargod  12.630804  75.093235
211               Ullody   671321  Kerala  Kasargod  12.562757  75.050106
212     Umda - padinhare   671319  Kerala  Kasargod  12.439982  75.016504
213               Uppala   671322  Kerala  Kasargod  12.675956  74.906262
214          Uppala Gate   671322  Kerala  Kasargod  12.682853  74.904801
215            Uppilakai   671314  Kerala  Kasargod  12.278937  75.124142
216                Urdur   671543  Kerala  Kasargod  12.549970  75.244989
217          Valiaparaba   671312  Kerala  Kasargod  12.508002  74.988211
218            Valiapoil   671313  Kerala  Kasargod  12.239143  75.184498
219          Vallikadavu   671533  Kerala  Kasargod  12.370397  75.354335
220            Vanin

### Importing and Installing Libraries

In [5]:
import numpy as np # library to handle data arrays
import pandas as pd # library for dealing tables
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json # library to handle JSON files

!conda install -c conda-forge geopy --yes  uncomment this line if you haven't completed the Foursquare API lab
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

!conda install -c conda-forge folium=0.5.0 --yes # uncomment this line if you haven't completed the Foursquare API lab
import folium # map rendering library

print('Libraries imported.')

/usr/bin/sh: -c: line 0: unexpected EOF while looking for matching `''
/usr/bin/sh: -c: line 1: syntax error: unexpected end of file
Solving environment: done

## Package Plan ##

  environment location: /opt/conda/envs/Python36

  added / updated specs: 
    - folium=0.5.0


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    vincent-0.4.4              |             py_1          28 KB  conda-forge
    openssl-1.1.1c             |       h516909a_0         2.1 MB  conda-forge
    branca-0.3.1               |             py_0          25 KB  conda-forge
    altair-3.2.0               |           py36_0         770 KB  conda-forge
    folium-0.5.0               |             py_0          45 KB  conda-forge
    ca-certificates-2019.9.11  |       hecc5488_0         144 KB  conda-forge
    certifi-2019.9.11          |           py36_0         147 KB  conda-forge
    -----------------------------

### Mapping Kanhangad

In [10]:
latitude = 12.3310887
longitude= 75.0915079999999

In [11]:
# create map of BARCELONA using latitude and longitude values above:
map_Kanhangad = folium.Map(location=[latitude, longitude], zoom_start=12)

# add markers to map
for lat, lng, label in zip(df_data_0['Latitude'], df_data_0['Longitude'], df_data_0['Location']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=4,
        popup=label,
        color='#a600ff',
        fill=True,
        fill_color='#ff0015',
        fill_opacity=0.5,
        parse_html=False).add_to(map_Kanhangad)  
    
map_Kanhangad

<a id='item5'></a>

## 5. Explore Venues in Kasaragod

Next, we are going to start utilizing the Foursquare API to explore the Places and segment them.

#### Define Foursquare Credentials and Version

In [12]:
CLIENT_ID = 'WN31QISG2QLOWAZMFUKF01G2RMCLU0NSMGZ5QUKNSHWM4TAQ' # my Foursquare ID
CLIENT_SECRET = '041BRMMYTUPSBFIVEEQBHEE1ED04WOZ2KZXYDSCSGXSVCAJF' # my Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: WN31QISG2QLOWAZMFUKF01G2RMCLU0NSMGZ5QUKNSHWM4TAQ
CLIENT_SECRET:041BRMMYTUPSBFIVEEQBHEE1ED04WOZ2KZXYDSCSGXSVCAJF


#### Now, let's get the top 100 venues that are in Kanhangad within a radius of 2000 meters.

In [15]:
LIMIT = 100 # limit of number of venues returned by Foursquare API
radius = 2000 # define radius

#### Let's create a function to repeat the same process to all the  Locations in Kasaragod

In [16]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Location', 
                  'Location Latitude', 
                  'Location Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category' ]
    
    return(nearby_venues)

#### Now write the code to run the above function on each Locations and create a new dataframe called *Kasaragod_venues*.

In [18]:
Kasaragod_venues = getNearbyVenues(names=df_data_0['Location'],
                                   latitudes=df_data_0['Latitude'],
                                   longitudes=df_data_0['Longitude']
                                    )

Achanthuruthi
Achikanam
Adkathbail
Adur
Ajanur
Alampady
Ammeri
Anandashrama
Anekallu
Angadimogaru
Arikady
Ariyapady
Attenganam
Badaje
Badur
Balal
Balanthode
Balemoole
Balla
Bandadka
Bangramanjeshwar
Bare
Bayar
Bedadka
Bedradka
Bekal
Bekur
Bela
Belluru
Bengalam
Beripadavu
Bheemanady
Bombrana
Chamnad
Chamundikunnu
Chandragiri
Charla
Chathamath
Chayoth
Chengala
Cheroor
Chervathur
Chimeni
Chippar
Chittari
Chittarikkal
Chully
Darkas
Delampady
Dharmathadka
Edachakai
Ednad
Edneer
Elambachi
Elerithattu
Erikulam
Haripuram
Heroor
Hidayathnagar
Hosabettu
Ichlangodu
Iriya
Iriyanni
Kadambar
Kaithakkad
Kakkebettu
Kalichanadukam
Kallakatta
Kallappally
Kalnad
Kanakappally
Kanathur
Kanhangad
Kanhangad South
Kanhiradkam
Kanhirapoil
Kannur
Karadka
Karindala
Karivedakam
Kasaragod
Kasaragod R s
Kattipoil
Kattukukke
Kayyaru
Kayyur
Keekan
Kidoor
Kilaikote
Kinanur
Kinningar
Kodiamme
Kodlamogaru
Kodoth
Kolathuri
Kolavayal
Koliyur
Kollampara
Konnakad
Kotakkat
Kotoor
Kottacherry
Kottamala Estate
Kottapuram
Kotto

#### Size of Kasaragod_venues and print first 16 rows.

In [19]:
print(Kasaragod_venues.shape)
Kasaragod_venues.head(15)

(137, 7)


Location  Location Latitude  Location Longitude  \
0   Anandashrama          12.339283           75.111810   
1      Ariyapady          12.508002           74.988211   
2      Ariyapady          12.508002           74.988211   
3      Ariyapady          12.508002           74.988211   
4      Ariyapady          12.508002           74.988211   
5      Ariyapady          12.508002           74.988211   
6      Ariyapady          12.508002           74.988211   
7          Bekal          12.420353           75.023146   
8          Bekal          12.420353           75.023146   
9       Bombrana          12.618196           74.945720   
10        Charla          12.699978           74.946158   
11    Chervathur          12.216516           75.162615   
12       Chimeni          12.242538           75.233321   
13      Chittari          12.360020           75.065495   
14        Darkas          12.508002           74.988211   

                             Venue  Venue Latitude  Venue Longitude  \
0              Indian Coffee House       12.338956        75.111428   
1                    Cochin bakery       12.507425        74.988241   
2                     Udupi Bhavan       12.507270        74.989359   
3                  Moviemax Circle       12.507574        74.988976   
4               SriKrishna Theatre       12.510904        74.989702   
5                     JK Residency       12.506781        74.991391   
6                  Kanyaka Theatre       12.504104        74.989878   
7        Kotikulam Railway Station       12.423585        75.023746   
8                        PalakunnÜ       12.423835        75.022099   
9                          Arikady       12.616864        74.942686   
10  Al Makkah Murgh Chanay and Bbq       12.696766        74.944997   
11                Pakkanar Theatre       12.215250        75.162373   
12                 City restaurant       12.238065        75.233489   
13                 New City Sweets       12.363093        75.063450   
14                   Cochin bakery       12.507425        74.988241   

         Venue Category  
0     Indian Restaurant  
1                Bakery  
2     Indian Restaurant  
3         Movie Theater  
4   Indie Movie Theater  
5                 Diner  
6   Indie Movie Theater  
7         Train Station  
8        Scenic Lookout  
9           Bus Station  
10            BBQ Joint  
11        Movie Theater  
12    Indian Restaurant  
13       Ice Cream Shop  
14               Bakery

#### Let's check how many venues were returned for each Locations

In [20]:

Kasaragod_venues.groupby('Location').count()

Location Latitude  Location Longitude  Venue  Venue Latitude  \
Location                                                                        
Anandashrama                     1                   1      1               1   
Ariyapady                        6                   6      6               6   
Bekal                            2                   2      2               2   
Bombrana                         1                   1      1               1   
Charla                           1                   1      1               1   
Chervathur                       1                   1      1               1   
Chimeni                          1                   1      1               1   
Chittari                         1                   1      1               1   
Darkas                           6                   6      6               6   
Hidayathnagar                    6                   6      6               6   
Iriya                            1                   1      1               1   
Kanhangad South                  1                   1      1               1   
Karindala                        6                   6      6               6   
Kasaragod                        6                   6      6               6   
Kasaragod R s                    6                   6      6               6   
Kattipoil                        6                   6      6               6   
Kodiamme                         6                   6      6               6   
Kotakkat                         6                   6      6               6   
Kottapuram                       1                   1      1               1   
Kumbla                           1                   1      1               1   
Majibail                         1                   1      1               1   
Manadka                          6                   6      6               6   
Manjeshwar                       2                   2      2               2   
Mogral                           3                   3      3               3   
Mogral Puthur                    2                   2      2               2   
Mulleria                         2                   2      2               2   
Muttomkadavil                    6                   6      6               6   
Nirchal                          1                   1      1               1   
Padnekkad                        1                   1      1               1   
Paraklayi                        6                   6      6               6   
Perla                            2                   2      2               2   
Pettikundu                       6                   6      6               6   
Pudukai                          6                   6      6               6   
Puthariadukkam                   6                   6      6               6   
Samekochi                        6                   6      6               6   
Talangara                        1                   1      1               1   
Uppala                           4                   4      4               4   
Uppala Gate                      2                   2      2               2   
Valiaparaba                      6                   6      6               6   
Vidyanagar                       2                   2      2               2   

                 Venue Longitude  Venue Category  
Location                                          
Anandashrama                   1               1  
Ariyapady                      6               6  
Bekal                          2               2  
Bombrana                       1               1  
Charla                         1               1  
Chervathur                     1               1  
Chimeni                        1               1  
Chittari                       1               1  
Darkas                         6               6  
Hidayathnagar                  6               6  
Ir

#### Let's find out how many unique categories can be curated from all the returned venues

In [22]:
print('The number of unique categories is {}.'.format(len(Kasaragod_venues['Venue Category'].unique())))

The number of unique categories is 26.


<a id='item1'></a>

## 6. Analyze Each Neighborhood

In [23]:
# one hot encoding
Kasaragod_onehot = pd.get_dummies(Kasaragod_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
Kasaragod_onehot['Location'] = Kasaragod_venues['Location'] 

# move district column to the first column
cols=list(Kasaragod_onehot.columns.values)
cols.pop(cols.index('Location'))
Kasaragod_onehot=Kasaragod_onehot[['Location']+cols]

# rename Neighborhood for Districts so that future merge works
Kasaragod_onehot.rename(columns = {'Location': 'Location'}, inplace = True)
Kasaragod_onehot.head(15)

Location  Arcade  Asian Restaurant  Athletics & Sports  BBQ Joint  \
0   Anandashrama       0                 0                   0          0   
1      Ariyapady       0                 0                   0          0   
2      Ariyapady       0                 0                   0          0   
3      Ariyapady       0                 0                   0          0   
4      Ariyapady       0                 0                   0          0   
5      Ariyapady       0                 0                   0          0   
6      Ariyapady       0                 0                   0          0   
7          Bekal       0                 0                   0          0   
8          Bekal       0                 0                   0          0   
9       Bombrana       0                 0                   0          0   
10        Charla       0                 0                   0          1   
11    Chervathur       0                 0                   0          0   
12       Chimeni       0                 0                   0          0   
13      Chittari       0                 0                   0          0   
14        Darkas       0                 0                   0          0   

    Bakery  Bus Station  Business Service  Clothing Store  Concert Hall  \
0        0            0                 0               0             0   
1        1            0                 0               0             0   
2        0            0                 0               0             0   
3        0            0                 0               0             0   
4        0            0                 0               0             0   
5        0            0                 0               0             0   
6        0            0                 0               0             0   
7        0            0                 0               0             0   
8        0            0                 0               0             0   
9        0            1                 0               0             0   
10       0            0                 0               0             0   
11       0            0                 0               0             0   
12       0            0                 0               0             0   
13       0            0                 0               0             0   
14       1            0                 0               0             0   

    Cruise  Currency Exchange  Diner  Football Stadium  Grocery Store  \
0        0                  0      0                 0              0   
1        0                  0      0                 0              0   
2        0                  0      0                 0              0   
3        0                  0      0                 0              0   
4        0                  0      0                 0              0   
5        0                  0      1                 0              0   
6        0                  0      0                 0              0   
7        0                  0      0                 0              0   
8        0                  0      0                 0              0   
9        0                  0      0                 0              0   
10       0                  0      0                 0              0   
11       0                  0      0                 0              0   
12       0                  0      0                 0              0   
13       0                  0      0                 0              0   
14       0                  0      0                 0              0   

    Ice Cream Shop  Indian Restaurant  Indie Movie Theater  Market  \
0                0                  1                    0       0   
1                0                  0                    0       0   
2                0                  1                    0       0   
3                0                  0                    0       0   
4                0                  0                  

In [24]:
Kasaragod_onehot.shape

(137, 27)

#### Next, let's group rows by neighborhood and by taking the mean of the frequency of occurrence of each category

In [35]:
Kasaragod_grouped = Kasaragod_onehot.groupby('Location').mean().reset_index()
Kasaragod_grouped

Location    Arcade  Asian Restaurant  Athletics & Sports  \
0      Anandashrama  0.000000          0.000000                 0.0   
1         Ariyapady  0.000000          0.000000                 0.0   
2             Bekal  0.000000          0.000000                 0.0   
3          Bombrana  0.000000          0.000000                 0.0   
4            Charla  0.000000          0.000000                 0.0   
5        Chervathur  0.000000          0.000000                 0.0   
6           Chimeni  0.000000          0.000000                 0.0   
7          Chittari  0.000000          0.000000                 0.0   
8            Darkas  0.000000          0.000000                 0.0   
9     Hidayathnagar  0.166667          0.166667                 0.0   
10            Iriya  0.000000          0.000000                 0.0   
11  Kanhangad South  0.000000          0.000000                 0.0   
12        Karindala  0.000000          0.000000                 0.0   
13        Kasaragod  0.000000          0.000000                 0.0   
14    Kasaragod R s  0.000000          0.000000                 0.0   
15        Kattipoil  0.000000          0.000000                 0.0   
16         Kodiamme  0.000000          0.000000                 0.0   
17         Kotakkat  0.000000          0.000000                 0.0   
18       Kottapuram  0.000000          0.000000                 0.0   
19           Kumbla  0.000000          0.000000                 0.0   
20         Majibail  0.000000          0.000000                 0.0   
21          Manadka  0.000000          0.000000                 0.0   
22       Manjeshwar  0.000000          0.000000                 0.0   
23           Mogral  0.000000          0.000000                 0.0   
24    Mogral Puthur  0.000000          0.000000                 0.5   
25         Mulleria  0.000000          0.000000                 0.0   
26    Muttomkadavil  0.000000          0.000000                 0.0   
27          Nirchal  0.000000          0.000000                 0.0   
28        Padnekkad  0.000000          0.000000                 0.0   
29        Paraklayi  0.000000          0.000000                 0.0   
30            Perla  0.000000          0.000000                 0.0   
31       Pettikundu  0.000000          0.000000                 0.0   
32          Pudukai  0.000000          0.000000                 0.0   
33   Puthariadukkam  0.000000          0.000000                 0.0   
34        Samekochi  0.000000          0.000000                 0.0   
35        Talangara  0.000000          1.000000                 0.0   
36           Uppala  0.250000          0.250000                 0.0   
37      Uppala Gate  0.000000          0.000000                 0.0   
38      Valiaparaba  0.000000          0.000000                 0.0   
39       Vidyanagar  0.000000          0.000000                 0.0   

    BBQ Joint    Bakery  Bus Station  Business Service  Clothing Store  \
0         0.0  0.000000          0.0               0.0             0.0   
1         0.0  0.166667          0.0               0.0             0.0   
2         0.0  0.000000          0.0               0.0             0.0   
3         0.0  0.000000          1.0               0.0             0.0   
4         1.0  0.000000          0.0               0.0             0.0   
5         0.0  0.000000          0.0               0.0             0.0   
6         0.0  0.000000          0.0               0.0             0.0   
7         0.0  0.000000          0.0               0.0             0.0   
8         0.0  0.166667          0.0               0.0             0.0   
9         0.0  0.000000          0.0               0.0             0.0   
10        0.0  0.000000          0.0               0.0             0.0   
11        0.0  0.000000          0.0               0.0             0.0   
12        0.0  0.166667          0.0               0.0             0.0   
13        0.0  0.166667          0.0               0.0         

#### Shape of Kasaragod_grouped

In [36]:
Kasaragod_grouped.shape

(40, 27)

#### Let's print each Locations along with the top 5 most common venues

In [37]:
num_top_venues = 5

for hood in Kasaragod_grouped['Location']:
    print("----"+hood+"----")
    temp = Kasaragod_grouped[Kasaragod_grouped['Location'] == hood].T.reset_index()
    temp.columns = ['venue','freq']
    temp = temp.iloc[1:]
    temp['freq'] = temp['freq'].astype(float)
    temp = temp.round({'freq': 2})
    print(temp.sort_values('freq', ascending=False).reset_index(drop=True).head(num_top_venues))
    print('\n')

----Anandashrama----
               venue  freq
0  Indian Restaurant   1.0
1             Arcade   0.0
2   Asian Restaurant   0.0
3      Train Station   0.0
4       Tennis Court   0.0


----Ariyapady----
                 venue  freq
0  Indie Movie Theater  0.33
1               Bakery  0.17
2        Movie Theater  0.17
3                Diner  0.17
4    Indian Restaurant  0.17


----Bekal----
              venue  freq
0     Train Station   0.5
1    Scenic Lookout   0.5
2            Arcade   0.0
3  Asian Restaurant   0.0
4      Tennis Court   0.0


----Bombrana----
            venue  freq
0     Bus Station   1.0
1          Arcade   0.0
2  Ice Cream Shop   0.0
3   Train Station   0.0
4    Tennis Court   0.0


----Charla----
            venue  freq
0       BBQ Joint   1.0
1          Arcade   0.0
2  Ice Cream Shop   0.0
3   Train Station   0.0
4    Tennis Court   0.0


----Chervathur----
              venue  freq
0     Movie Theater   1.0
1            Arcade   0.0
2  Asian Restaurant   0.0
3 

#### First, let's write a function to sort the venues in descending order.

In [38]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

#### Now let's create the new dataframe and display the top 10 venues for each Locations.

In [39]:
num_top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Location']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Location'] = Kasaragod_grouped['Location']

for ind in np.arange(Kasaragod_grouped.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(Kasaragod_grouped.iloc[ind, :], num_top_venues)

neighborhoods_venues_sorted.shape

(40, 11)

<a id='item4'></a>

## 7. Cluster Neighborhoods

Run *k*-means to cluster the neighborhood into 7 clusters.

In [40]:
# set number of clusters
kclusters = 7

Kasaragod_grouped_clustering = Kasaragod_grouped.drop('Location', 1)
# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=1).fit(Kasaragod_grouped_clustering)

# check cluster labels generated for each row in the dataframe
print(kmeans.labels_[0:10])
print(len(kmeans.labels_))

[1 0 2 3 6 0 1 2 0 2]
40


#### Let's create a new dataframe that includes the cluster as well as the top 10 venues for each Locations.

In [41]:
# add clustering labels
neighborhoods_venues_sorted.insert(0, 'Cluster Labels', kmeans.labels_)

Kasaragod_merged = df_data_0

# merge Kasaragod_grouped with toronto_data to add latitude/longitude for each neighborhood
Kasaragod_merged  = Kasaragod_merged .join(neighborhoods_venues_sorted.set_index('Location'), on='Location')



In [42]:
Kasaragod_merged

Location  Pincode   State  District   Latitude  Longitude  \
0          Achanthuruthi   671351  Kerala  Kasargod  12.229580  75.123890   
1              Achikanam   671531  Kerala  Kasargod  12.332950  75.133900   
2             Adkathbail   671121  Kerala  Kasargod  12.514748  74.977047   
3                   Adur   671543  Kerala  Kasargod  12.557546  75.180804   
4                 Ajanur   671531  Kerala  Kasargod  12.329081  75.074803   
5               Alampady   671123  Kerala  Kasargod  12.534245  75.040736   
6                 Ammeri   671322  Kerala  Kasargod  12.714228  74.999097   
7           Anandashrama   671531  Kerala  Kasargod  12.339283  75.111810   
8               Anekallu   671323  Kerala  Kasargod  12.752339  74.993968   
9           Angadimogaru   671321  Kerala  Kasargod  12.635864  75.014521   
10               Arikady   671321  Kerala  Kasargod  12.611301  74.947181   
11             Ariyapady   671551  Kerala  Kasargod  12.508002  74.988211   
12            Attenganam   671531  Kerala  Kasargod  12.389619  75.196852   
13                Badaje   671323  Kerala  Kasargod  12.731822  74.906262   
14                 Badur   671321  Kerala  Kasargod  12.658037  75.031909   
15                 Balal   671533  Kerala  Kasargod  12.392013  75.265406   
16            Balanthode   671532  Kerala  Kasargod  12.459245  75.320811   
17             Balemoole   671552  Kerala  Kasargod  12.668430  75.144999   
18                 Balla   671531  Kerala  Kasargod  12.320475  75.106612   
19              Bandadka   671541  Kerala  Kasargod  12.501006  75.268323   
20      Bangramanjeshwar   671323  Kerala  Kasargod  12.709134  74.900416   
21                  Bare   671319  Kerala  Kasargod  12.447905  75.042131   
22                 Bayar   671348  Kerala  Kasargod  12.691564  75.031909   
23               Bedadka   671541  Kerala  Kasargod  12.456411  75.169131   
24              Bedradka   671124  Kerala  Kasargod  12.450110  75.150031   
25                 Bekal   671318  Kerala  Kasargod  12.420353  75.023146   
26                 Bekur   671322  Kerala  Kasargod  12.672056  74.935491   
27                  Bela   671321  Kerala  Kasargod  12.569345  75.043592   
28               Belluru   671543  Kerala  Kasargod  12.586296  75.170590   
29              Bengalam   671314  Kerala  Kasargod  12.293883  75.153036   
30            Beripadavu   671322  Kerala  Kasargod  12.710432  75.053069   
31            Bheemanady   671314  Kerala  Kasargod  12.326372  75.288737   
32              Bombrana   671321  Kerala  Kasargod  12.618196  74.945720   
33               Chamnad   671317  Kerala  Kasargod  12.489235  75.008540   
34         Chamundikunnu   671532  Kerala  Kasargod  12.475350  75.309148   
35           Chandragiri   671317  Kerala  Kasargod  12.469937  75.003245   
36                Charla   671323  Kerala  Kasargod  12.699978  74.946158   
37            Chathamath   671314  Kerala  Kasargod  12.247964  75.155688   
38               Chayoth   671314  Kerala  Kasargod  12.284223  75.180804   
39              Chengala   671541  Kerala  Kasargod  12.502372  75.043592   
40               Cheroor   671123  Kerala  Kasargod  12.499418  75.040671   
41            Chervathur   671313  Kerala  Kasargod  12.216516  75.162615   
42               Chimeni   671313  Kerala  Kasargod  12.242538  75.233321   
43               Chippar   671322  Kerala  Kasargod  12.704027  75.005619   
44              Chittari   671316  Kerala  Kasargod  12.360020  75.065495   
45          Chittarikkal   671326  Kerala  Kasargod  12.321988  75.361622   
46                Chully   671533  Kerala  Kasargod  12.386692  75.320811   
47                Darkas   671533  Kerala  Kasargod  12.508002  74.988211   
48             Delampady   671543  Kerala  Kasargod  12.597277  75.291653   
49          Dharmathadka   671324  Kerala  Kasargod  12.666251  75.028988   
50             Edachakai   671310  Kerala  Kasargod  12.166758  75.158186   
51          

#### Removing NaN Rows

In [43]:
Kasaragod_merged.dropna()

Location  Pincode   State  District   Latitude  Longitude  \
7       Anandashrama   671531  Kerala  Kasargod  12.339283  75.111810   
11         Ariyapady   671551  Kerala  Kasargod  12.508002  74.988211   
25             Bekal   671318  Kerala  Kasargod  12.420353  75.023146   
32          Bombrana   671321  Kerala  Kasargod  12.618196  74.945720   
36            Charla   671323  Kerala  Kasargod  12.699978  74.946158   
41        Chervathur   671313  Kerala  Kasargod  12.216516  75.162615   
42           Chimeni   671313  Kerala  Kasargod  12.242538  75.233321   
44          Chittari   671316  Kerala  Kasargod  12.360020  75.065495   
47            Darkas   671533  Kerala  Kasargod  12.508002  74.988211   
58     Hidayathnagar   671123  Kerala  Kasargod  12.673660  74.908820   
61             Iriya   671531  Kerala  Kasargod  12.390790  75.167671   
73   Kanhangad South   671531  Kerala  Kasargod  12.291546  75.100097   
78         Karindala   671314  Kerala  Kasargod  12.508002  74.988211   
80         Kasaragod   671121  Kerala  Kasargod  12.508002  74.988211   
81     Kasaragod R s   671121  Kerala  Kasargod  12.508002  74.988211   
82         Kattipoil   671314  Kerala  Kasargod  12.508002  74.988211   
91          Kodiamme   671321  Kerala  Kasargod  12.508002  74.988211   
99          Kotakkat   671357  Kerala  Kasargod  12.508002  74.988211   
103       Kottapuram   671341  Kerala  Kasargod  12.232985  75.127735   
108           Kumbla   671321  Kerala  Kasargod  12.594554  74.947181   
118         Majibail   671323  Kerala  Kasargod  12.693856  74.901877   
121          Manadka   671541  Kerala  Kasargod  12.508002  74.988211   
127       Manjeshwar   671323  Kerala  Kasargod  12.724554  74.885011   
131           Mogral   671321  Kerala  Kasargod  12.572562  74.953026   
132    Mogral Puthur   671124  Kerala  Kasargod  12.553530  74.961792   
138         Mulleria   671543  Kerala  Kasargod  12.550978  75.163294   
141    Muttomkadavil   671533  Kerala  Kasargod  12.508002  74.988211   
147          Nirchal   671321  Kerala  Kasargod  12.588388  75.034829   
153        Padnekkad   671328  Kerala  Kasargod  12.265525  75.114082   
164        Paraklayi   671531  Kerala  Kasargod  12.508002  74.988211   
175            Perla   671552  Kerala  Kasargod  12.642642  75.104914   
179       Pettikundu   671313  Kerala  Kasargod  12.508002  74.988211   
183          Pudukai   671314  Kerala  Kasargod  12.508002  74.988211   
186   Puthariadukkam   671314  Kerala  Kasargod  12.508002  74.988211   
192        Samekochi   671541  Kerala  Kasargod  12.508002  74.988211   
196        Talangara   671122  Kerala  Kasargod  12.484142  74.992472   
213           Uppala   671322  Kerala  Kasargod  12.675956  74.906262   
214      Uppala Gate   671322  Kerala  Kasargod  12.682853  74.904801   
217      Valiaparaba   671312  Kerala  Kasargod  12.508002  74.988211   
224       Vidyanagar   671123  Kerala  Kasargod  12.514633  75.019568   

     Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
7               1.0     Indian Restaurant      Udupi Restaurant   
11              0.0   Indie Movie Theater                Bakery   
25              2.0        Scenic Lookout         Train Station   
32              3.0           Bus Station      Udupi Restaurant   
36              6.0             BBQ Joint      Udupi Restaurant   
41              0.0         Movie Theater      Udupi Restaurant   
42              1.0     Indian Restaurant      Udupi Restaurant   
44              2.0        Ice Cream Shop      Udupi Restaurant   
47              0.0   Indie Movie Theater                Bakery   
58              2.0                Arcade      Asian Restaurant   
61              1.0     Indian Restaurant      Udupi Restaurant   
73              4.0       Motorcycle Shop      Udupi Restaurant   
78              0.0   Indie Movie Theater                Bakery   
80              0.0   Indie Movie Theater                Bakery   
81           

#### Finally, let's visualize the resulting clusters

In [44]:
map_clusters = folium.Map(location=[latitude, longitude], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i+x+(i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(Kasaragod_merged['Latitude'], Kasaragod_merged['Longitude'], Kasaragod_merged['Location'],kmeans.labels_):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=3,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

## 8. Examine Clusters

#### Now, you can examine each cluster and determine the discriminating venue categories that distinguish each cluster.

#### Cluster 1

In [45]:
Kasaragod_merged.loc[Kasaragod_merged['Cluster Labels'] == 0, Kasaragod_merged.columns[[0] + list(range(5, Kasaragod_merged.shape[1]))]]

Location  Longitude  Cluster Labels 1st Most Common Venue  \
11        Ariyapady  74.988211             0.0   Indie Movie Theater   
41       Chervathur  75.162615             0.0         Movie Theater   
47           Darkas  74.988211             0.0   Indie Movie Theater   
78        Karindala  74.988211             0.0   Indie Movie Theater   
80        Kasaragod  74.988211             0.0   Indie Movie Theater   
81    Kasaragod R s  74.988211             0.0   Indie Movie Theater   
82        Kattipoil  74.988211             0.0   Indie Movie Theater   
91         Kodiamme  74.988211             0.0   Indie Movie Theater   
99         Kotakkat  74.988211             0.0   Indie Movie Theater   
118        Majibail  74.901877             0.0                 Diner   
121         Manadka  74.988211             0.0   Indie Movie Theater   
131          Mogral  74.953026             0.0      Football Stadium   
141   Muttomkadavil  74.988211             0.0   Indie Movie Theater   
164       Paraklayi  74.988211             0.0   Indie Movie Theater   
179      Pettikundu  74.988211             0.0   Indie Movie Theater   
183         Pudukai  74.988211             0.0   Indie Movie Theater   
186  Puthariadukkam  74.988211             0.0   Indie Movie Theater   
192       Samekochi  74.988211             0.0   Indie Movie Theater   
217     Valiaparaba  74.988211             0.0   Indie Movie Theater   
224      Vidyanagar  75.019568             0.0            Restaurant   

    2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
11                 Bakery         Movie Theater     Indian Restaurant   
41       Udupi Restaurant                 Diner      Asian Restaurant   
47                 Bakery         Movie Theater     Indian Restaurant   
78                 Bakery         Movie Theater     Indian Restaurant   
80                 Bakery         Movie Theater     Indian Restaurant   
81                 Bakery         Movie Theater     Indian Restaurant   
82                 Bakery         Movie Theater     Indian Restaurant   
91                 Bakery         Movie Theater     Indian Restaurant   
99                 Bakery         Movie Theater     Indian Restaurant   
118      Udupi Restaurant         Train Station      Asian Restaurant   
121                Bakery         Movie Theater     Indian Restaurant   
131                Bakery     Indian Restaurant                 Diner   
141                Bakery         Movie Theater     Indian Restaurant   
164                Bakery         Movie Theater     Indian Restaurant   
179                Bakery         Movie Theater     Indian Restaurant   
183                Bakery         Movie Theater     Indian Restaurant   
186                Bakery         Movie Theater     Indian Restaurant   
192                Bakery         Movie Theater     Indian Restaurant   
217                Bakery         Movie Theater     Indian Restaurant   
224                Bakery      Udupi Restaurant                 Diner   

    5th Most Common Venue 6th Most Common Venue 7th Most Common Venue  \
11                  Diner      Udupi Restaurant     Currency Exchange   
41     Athletics & Sports             BBQ Joint                Bakery   
47                  Diner      Udupi Restaurant     Currency Exchange   
78                  Diner      Udupi Restaurant     Currency Exchange   
80                  Diner      Udupi Restaurant     Currency Exchange   
81                  Diner      Udupi Restaurant     Currency Exchange   
82                  Diner      Udupi Restaurant     Currency Exchange   
91                  Diner      Udupi Restaurant     Currency Exchange   
99                  Diner      Udupi Restaurant     Currency Exchange   
118    Athletics & Sports             BBQ Joint                Bakery   
121                 Diner      Udupi Restaurant     Currency Exchange   
131      Asian Restaurant    Athletics & Sports             BBQ Joint   
141            

#### Cluster 2

In [52]:
Kasaragod_merged.loc[Kasaragod_merged['Cluster Labels'] == 1, Kasaragod_merged.columns[[0] + list(range(5, Kasaragod_merged.shape[1]))]]

Location  Longitude  Cluster Labels 1st Most Common Venue  \
7   Anandashrama  75.111810             1.0     Indian Restaurant   
42       Chimeni  75.233321             1.0     Indian Restaurant   
61         Iriya  75.167671             1.0     Indian Restaurant   

   2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
7       Udupi Restaurant                 Diner      Asian Restaurant   
42      Udupi Restaurant                 Diner      Asian Restaurant   
61      Udupi Restaurant                 Diner      Asian Restaurant   

   5th Most Common Venue 6th Most Common Venue 7th Most Common Venue  \
7     Athletics & Sports             BBQ Joint                Bakery   
42    Athletics & Sports             BBQ Joint                Bakery   
61    Athletics & Sports             BBQ Joint                Bakery   

   8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
7            Bus Station      Business Service         Clothing Store  
42           Bus Station      Business Service         Clothing Store  
61           Bus Station      Business Service         Clothing Store

#### Cluster 3

In [55]:
Kasaragod_merged.loc[Kasaragod_merged['Cluster Labels'] == 2, Kasaragod_merged.columns[[0] + list(range(5, Kasaragod_merged.shape[1]))]]

Location  Longitude  Cluster Labels 1st Most Common Venue  \
25           Bekal  75.023146             2.0        Scenic Lookout   
44        Chittari  75.065495             2.0        Ice Cream Shop   
58   Hidayathnagar  74.908820             2.0                Arcade   
103     Kottapuram  75.127735             2.0                Cruise   
108         Kumbla  74.947181             2.0         Train Station   
127     Manjeshwar  74.885011             2.0         Train Station   
132  Mogral Puthur  74.961792             2.0          Tennis Court   
153      Padnekkad  75.114082             2.0      Business Service   
175          Perla  75.104914             2.0                Market   
196      Talangara  74.992472             2.0      Asian Restaurant   
213         Uppala  74.906262             2.0                Arcade   
214    Uppala Gate  74.904801             2.0          Concert Hall   

    2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
25          Train Station      Udupi Restaurant                 Diner   
44       Udupi Restaurant                 Diner      Asian Restaurant   
58       Asian Restaurant              Pharmacy          Concert Hall   
103      Udupi Restaurant         Train Station      Asian Restaurant   
108      Udupi Restaurant      Asian Restaurant    Athletics & Sports   
127           Bus Station      Udupi Restaurant      Asian Restaurant   
132    Athletics & Sports      Udupi Restaurant                 Diner   
153      Udupi Restaurant         Train Station      Asian Restaurant   
175        Clothing Store      Udupi Restaurant                 Diner   
196      Udupi Restaurant         Train Station    Athletics & Sports   
213      Asian Restaurant          Concert Hall         Grocery Store   
214     Currency Exchange      Udupi Restaurant         Train Station   

    5th Most Common Venue 6th Most Common Venue 7th Most Common Venue  \
25       Asian Restaurant    Athletics & Sports             BBQ Joint   
44     Athletics & Sports             BBQ Joint                Bakery   
58          Grocery Store         Train Station     Currency Exchange   
103    Athletics & Sports             BBQ Joint                Bakery   
108             BBQ Joint                Bakery           Bus Station   
127    Athletics & Sports             BBQ Joint                Bakery   
132      Asian Restaurant             BBQ Joint                Bakery   
153    Athletics & Sports             BBQ Joint                Bakery   
175      Asian Restaurant    Athletics & Sports             BBQ Joint   
196             BBQ Joint                Bakery           Bus Station   
213                 Diner    Athletics & Sports             BBQ Joint   
214      Asian Restaurant    Athletics & Sports             BBQ Joint   

    8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
25                 Bakery           Bus Station       Business Service  
44            Bus Station      Business Service         Clothing Store  
58     Athletics & Sports             BBQ Joint                 Bakery  
103           Bus Station      Business Service         Clothing Store  
108      Business Service        Clothing Store           Concert Hall  
127      Business Service        Clothing Store           Concert Hall  
132           Bus Station      Business Service         Clothing Store  
153           Bus Station        Clothing Store           Concert Hall  
175                Bakery           Bus Station       Business Service  
196      Business Service        Clothing Store           Concert Hall  
213                Bakery           Bus Station       Business Service  
214                Bakery           Bus Station       Business Service

#### Cluster 4

In [54]:
Kasaragod_merged.loc[Kasaragod_merged['Cluster Labels'] == 3, Kasaragod_merged.columns[[0] + list(range(5, Kasaragod_merged.shape[1]))]]

Location  Longitude  Cluster Labels 1st Most Common Venue  \
32  Bombrana   74.94572             3.0           Bus Station   

   2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
32      Udupi Restaurant         Train Station      Asian Restaurant   

   5th Most Common Venue 6th Most Common Venue 7th Most Common Venue  \
32    Athletics & Sports             BBQ Joint                Bakery   

   8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
32      Business Service        Clothing Store           Concert Hall

#### Cluster 5

In [49]:
Kasaragod_merged.loc[Kasaragod_merged['Cluster Labels'] == 4, Kasaragod_merged.columns[[0] + list(range(5, Kasaragod_merged.shape[1]))]]

Location  Longitude  Cluster Labels 1st Most Common Venue  \
73  Kanhangad South  75.100097             4.0       Motorcycle Shop   

   2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
73      Udupi Restaurant                 Diner      Asian Restaurant   

   5th Most Common Venue 6th Most Common Venue 7th Most Common Venue  \
73    Athletics & Sports             BBQ Joint                Bakery   

   8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
73           Bus Station      Business Service         Clothing Store

#### Cluster 6

In [50]:
Kasaragod_merged.loc[Kasaragod_merged['Cluster Labels'] == 5, Kasaragod_merged.columns[[0] + list(range(5, Kasaragod_merged.shape[1]))]]

Location  Longitude  Cluster Labels 1st Most Common Venue  \
147  Nirchal  75.034829             5.0      Udupi Restaurant   

    2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
147         Train Station      Asian Restaurant    Athletics & Sports   

    5th Most Common Venue 6th Most Common Venue 7th Most Common Venue  \
147             BBQ Joint                Bakery           Bus Station   

    8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
147      Business Service        Clothing Store           Concert Hall

#### Cluster 7

In [51]:
Kasaragod_merged.loc[Kasaragod_merged['Cluster Labels'] == 6, Kasaragod_merged.columns[[0] + list(range(5, Kasaragod_merged.shape[1]))]]

Location  Longitude  Cluster Labels 1st Most Common Venue  \
36     Charla  74.946158             6.0             BBQ Joint   
138  Mulleria  75.163294             6.0             BBQ Joint   

    2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
36       Udupi Restaurant         Train Station      Asian Restaurant   
138        Clothing Store      Udupi Restaurant         Train Station   

    5th Most Common Venue 6th Most Common Venue 7th Most Common Venue  \
36     Athletics & Sports                Bakery           Bus Station   
138      Asian Restaurant    Athletics & Sports                Bakery   

    8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
36       Business Service        Clothing Store           Concert Hall  
138           Bus Station      Business Service           Concert Hall

<a id='item9'></a>

## 9. Results




Analysing the Clusters and Places, Bekal with Latitude 12.420353 and Longitude 75.023146 best place for Establishing Hostel for students, Apartment for Professors and Govt. Medical College Hospital. Since Scenic Lookout is the first most common venue. The Nature-based intervention will improve mental health(Reference: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5663019/). That will improve the mood of students and teachers. Most of the other venues are almost common in all places. This is the best place for starting an Academic campus. It is connected via, Rail, Road. So, Patients in the District can be easily accessed. Bekal is in the middle of the district. So, People in the Kasaragod district can utilize the facility equally. I recommend this place for the Govt. College Project.

This notebook was created by [Maneesh Narayanan](https://www.linkedin.com/in/maneesh-narayanan-545581138/)